app/controllers/base_resource_controller.rb

```diff
@@ -46,7 +46,7 @@ def browse_everything_params
   def selected_files
     files = browse_everything_params.fetch(:selected_files, {})
 
-    files.values.map { |value| BrowseEverything::Resource.new(value) }
+    files.values.map { |value| BrowseEverything::Resource.new(value) }.uniq
```

Comment:
Could you do the `uniq` before the `map` and avoid creating the duplicate objects?